File: PrinterAgent Demos/RemotePrinterSDK.thinrdp.64/demos/RemotePrinterDemoNet/MainForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using Cybele.Thinfinity;

namespace RemotePrinterDemoNet
{
    public partial class MainForm : Form
    {
        private RemotePrinter remotePrinter;
        public MainForm()
        {
            remotePrinter = new RemotePrinter();
            InitializeComponent();
        }

        private void BtnGetPrinters_Click(object sender, EventArgs e)
        {
            string printers;
            string errMsg;
            int errCode;
            if (!remotePrinter.GetPrinters(";", out printers)) {
                remotePrinter.LastError(out errCode, out errMsg);
                MessageBox.Show(errMsg);
                return;
            }
            String[] elements = printers.Split(';');
            LBoxPrinters.Items.Clear();
            foreach (var element in elements)
            {
                LBoxPrinters.Items.Add(element);
            }
        }

        private void LBoxPrinters_SelectedIndexChanged(object sender, EventArgs e)
        {
            txtPrinterName.Text = LBoxPrinters.SelectedItem.ToString();
        }

        private void BtnPrintDoc_Click(object sender, EventArgs e)
        {
            bool printing = false, done = false;
            string errMsg, docID;
            int errCode;

            if (remotePrinter.BeginDoc((int)PrintType.PRINT_TYPE_RAW, txtPrinterName.Text, txtDocTitle.Text, (int)Encode.PRINT_ENCODE_UTF8, out docID))
            {
                printing = true;

                if (remotePrinter.Print(docID,textBox1.Text))
                {
                    if (remotePrinter.EndDoc(docID))
                    {
                        done = true;
                        printing = false;
                        MessageBox.Show("done");
                    }
                }

                if (!done)
                {
                    remotePrinter.LastError(out errCode, out errMsg);
                    MessageBox.Show(errMsg);
                }

                if (printing)
                {
                    if (!remotePrinter.Abort(docID))
                    {
                        remotePrinter.LastError(out errCode, out errMsg);
                        MessageBox.Show(errMsg);
                    }
                }

            }
        }
    }
}

File: PrinterAgent Demos/RemotePrinterSDK.thinrdp.32/demos/RemotePrinterDemoVBNet/MainForm.vb
Public Class MainForm
    Private remotePrinter As Cybele.Thinfinity.RemotePrinter
    Private Sub MainForm_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        remotePrinter = New Cybele.Thinfinity.RemotePrinter()
    End Sub

    Private Sub LBoxPrinters_SelectedValueChanged(sender As Object, e As EventArgs) Handles LBoxPrinters.SelectedValueChanged

        txtPrinterName.Text = LBoxPrinters.SelectedItem.ToString()

    End Sub

    Private Sub BtnGetPrinters_Click(sender As Object, e As EventArgs) Handles BtnGetPrinters.Click
        Dim errCode As Integer = 0
        Dim errMsg As String = String.Empty
        Dim printers As String = String.Empty

        If Not remotePrinter.GetPrinters(";", printers) Then

            remotePrinter.LastError(errCode, errMsg)
            MessageBox.Show(errMsg)
            Return
        End If

        Dim elements() As String = printers.Split(";")
        LBoxPrinters.Items.Clear()
        For Each element As String In elements
            LBoxPrinters.Items.Add(element)
        Next
    End Sub

    Private Sub BtnPrintDoc_Click(sender As Object, e As EventArgs) Handles BtnPrintDoc.Click
        Dim Printing As Boolean = False
        Dim Done As Boolean = False
        Dim ErrMsg As String = String.Empty
        Dim ErrCode As Integer = 0
        Dim DocID As String = String.Empty

        If remotePrinter.BeginDoc(CInt(Cybele.Thinfinity.PrintType.PRINT_TYPE_RAW),
                                  txtPrinterName.Text, txtDocTitle.Text,
                                  CInt(Cybele.Thinfinity.Encode.PRINT_ENCODE_UTF8), DocID) Then
            Printing = True
            If remotePrinter.Print(DocID, txtData.Text) Then
                If remotePrinter.EndDoc(DocID) Then
                    Done = True
                    Printing = False
                    MessageBox.Show("done.")
                End If
            End If
        End If

        If Not Done Then
            remotePrinter.LastError(ErrCode, ErrMsg)
            MessageBox.Show(ErrMsg)
        End If

        If Printing Then
            If Not remotePrinter.Abort(DocID) Then
                remotePrinter.LastError(ErrCode, ErrMsg)
                MessageBox.Show(ErrMsg)
            End If
        End If

    End Sub
End Class
